editor/src/components/canvas/controls/select-mode/remix-scene-label.tsx

```diff
@@ -0,0 +1,283 @@
+): string => `${EP.toString(path)}-remix-scene-label-button-${button}`
+
+interface RemixSceneLabelControlProps {
+  maybeHighlightOnHover: (target: ElementPath) => void
+  maybeClearHighlightsOnHoverEnd: () => void
+}
+
+interface RemixSceneLabelProps extends RemixSceneLabelControlProps {
+  target: ElementPath
+}
+
+export const RemixSceneLabelTestID = (path: ElementPath): string =>
+  `${EP.toString(path)}-remix-scene-label`
+
+export const RemixSceneLabelControl = React.memo<RemixSceneLabelControlProps>((props) => {
+  const sceneTargets = useEditorState(
+    Substores.metadata,
+    (store) =>
+      Object.values(store.editor.jsxMetadata).filter((element) =>
+        MetadataUtils.isProbablyRemixScene(store.editor.jsxMetadata, element.elementPath),
+      ),
+    'RemixSceneLabelControl',
+  )
+  return (
+    <>
+      {sceneTargets.map((element) => (
+        <RemixSceneLabel
+          key={EP.toString(element.elementPath)}
+          target={element.elementPath}
+          maybeHighlightOnHover={props.maybeHighlightOnHover}
+          maybeClearHighlightsOnHoverEnd={props.maybeClearHighlightsOnHoverEnd}
+        />
+      ))}
+    </>
+  )
+})
+
+const RemixSceneLabel = React.memo<RemixSceneLabelProps>((props) => {
+  const colorTheme = useColorTheme()
+  const dispatch = useDispatch()
+
+  const [navigationData] = useAtom(RemixNavigationAtom)
+
+  const currentPath = (navigationData[EP.toString(props.target)] ?? null)?.location.pathname
+  const isIndexRoute = currentPath === '/'
+
+  const label = isIndexRoute ? RemixSceneHomeLabel : currentPath
+
+  const forward = React.useCallback(
+    () => navigationData[EP.toString(props.target)]?.forward(),
+    [navigationData, props.target],
+  )
+  const back = React.useCallback(
+    () => navigationData[EP.toString(props.target)]?.back(),
+    [navigationData, props.target],
+  )
+  const home = React.useCallback(
+    () => navigationData[EP.toString(props.target)]?.home(),
+    [navigationData, props.target],
+  )
+
+  const labelSelectable = useEditorState(
+    Substores.restOfEditor,
+    (store) => !store.editor.keysPressed['z'],
+    'RemixSceneLabel Z key pressed',
+  )
+
+  const frame = useEditorState(
+    Substores.metadata,
+    (store) => MetadataUtils.getFrameInCanvasCoords(props.target, store.editor.jsxMetadata),
+    'RemixSceneLabel frame',
+  )
+
+  const canvasOffset = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.realCanvasOffset,
+    'RemixSceneLabel canvasOffset',
+  )
+  const scale = useEditorState(
+    Substores.canvas,
+    (store) => store.editor.canvas.scale,
+    'RemixSceneLabel scale',
+  )
+  const baseFontSize = 9
+  const scaledFontSize = baseFontSize / scale
+  const scaledLineHeight = 17 / scale
+  const paddingY = scaledFontSize / 2
+  const offsetY = scaledFontSize
+  const offsetX = scaledFontSize
+  const borderRadius = 3 / scale
+
+  const storeRef = useRefEditorState((store) => {
```

Comment:
Is there a reason why this is not just an editorModeRef?

Reply:
It was the same in `SceneLabel`, renamed it in https://github.com/concrete-utopia/utopia/pull/4195/commits/d320a9a66381c35564cbdacef4baf60a98de2ae9